### ISSUE TITLE
feature: read input data from stdin

### ISSUE THREAD

**Issue by hbschr** [bbc/audiowaveform#13]
it would be nice to build pipes like this:

```
ffmpeg -i <movie_file> -f wav -  |  audiowaveform -i - -o file.json
```

without having to tmpfile-dump the wav file.

**Comment by oncletom** [comment]
It would be a great use case for #28 — although I suppose the stdin/stdout formats (dat, json etc.) should be provided in a different way, as it cannot be read from the filename extension.

**Comment by pkeuter** [comment]
+1

**Comment by cryptix** [comment]
+1 `--input-format=(wav|mp3)` and `--output-format=(dat|json)` maybe?

**Comment by chrisn** [comment]
I already have work-in-progress code that adds `--input-format` and `--output-format` options as suggested by @cryptix, so I'll try to make some time to finish this up.

**Comment by starak** [comment]
+1

**Comment by zaytsevfuu** [comment]
+1 ;)

**Comment by drrainlab** [comment]
+1 
It would be great

**Comment by zenondoh** [comment]
+1
It's a cool suggestion.

**Comment by nywhere** [comment]
+1
Would be definitely great

**Comment by joneslee85** [comment]
Someone has implemented reading from stdin https://github.com/bbc/audiowaveform/commit/277e625c66f6675ceda56de6be132d12656027ee

**Comment by chrisn** [comment]
Please take a look at the `stdio2` branch and https://github.com/bbc/audiowaveform/pull/89. Feedback welcome!

**Comment by chrisn** [comment]
I think PR #89 is ready to merge. Please take a look and let me know if this works for you or if there are any issues. Thanks!

**Comment by chrisn** [comment]
This is implemented in v1.4.0. Feedback welcome!
